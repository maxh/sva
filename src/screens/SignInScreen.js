import React, { Component } from 'react';
import {
  StyleSheet,
  Text,
  View,
} from 'react-native';
import { connect } from 'react-redux';
import Icon from 'react-native-vector-icons/FontAwesome';
import { signIn } from '../actions/auth';

const BLUE = '#007AFF';
const WHITE = '#FFFFFF';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: BLUE,
  },
  title: {
    fontSize: 50,
    textAlign: 'center',
    margin: 5,
    color: WHITE,
  },
  subtitle: {
    fontSize: 20,
    textAlign: 'center',
    marginBottom: 50,
    color: WHITE,
  },
});

export class SignInScreen extends Component {
  static navigatorStyle = {
    drawUnderNavBar: true,
    navBarHidden: true,
  };

  render() {
    let inner;
    if (this.props.isLoading) {
      inner = <Text>Loading...</Text>;
    } else {
      inner = (
        <View>
          <Text style={styles.title}>Scout</Text>
          <Text style={styles.subtitle}>Your voice companion</Text>
          <Icon.Button
            name="google"
            color="#000000"
            backgroundColor="#FFFFFF"
            onPress={this.props.signIn}
          >
            Login with Google
          </Icon.Button>
        </View>
      );
    }
    return (
      <View style={styles.container}>
        {inner}
      </View>
    );
  }
}

SignInScreen.propTypes = {
  signIn: React.PropTypes.func.isRequired,
  isLoading: React.PropTypes.bool.isRequired,
};

const mapStateToProps = state => ({
  isLoading: state.auth.googleUser.isLoading || state.auth.deviceToken.isLoading,
});

const mapDispatchToProps = { signIn };

export default connect(mapStateToProps, mapDispatchToProps)(SignInScreen);
