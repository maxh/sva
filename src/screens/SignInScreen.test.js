import 'react-native';
import { shallow } from 'enzyme';
import React from 'react';
import { SignInScreen } from './SignInScreen';

jest.mock('react-native-google-signin', () => 'signin');


it('renders correctly when not loading', () => {
  const props = {
    signIn: jest.fn(),
    isLoading: false
  };

  const tree = shallow(
    <SignInScreen {...props} />
  );

  expect(tree).toMatchSnapshot();
});


it('renders correctly when loading', () => {
  const props = {
    signIn: jest.fn(),
    isLoading: true
  };

  const tree = shallow(
    <SignInScreen {...props} />
  );

  expect(tree).toMatchSnapshot();
});
