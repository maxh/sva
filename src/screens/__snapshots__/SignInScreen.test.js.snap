exports[`test renders correctly when loading 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "alignItems": "center",
            "backgroundColor": "#007AFF",
            "flex": 1,
            "justifyContent": "center",
          }
    }>
    <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail">
        Loading...
    </Text>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "alignItems": "center",
                  "backgroundColor": "#007AFF",
                  "flex": 1,
                  "justifyContent": "center",
                }
      }>
      <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail">
            Loading...
      </Text>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignInScreen
        isLoading={true}
        signIn={[Function]} />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignInScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isLoading": true,
          "signIn": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#007AFF",
                              "flex": 1,
                              "justifyContent": "center",
                            }
          }>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail">
                    Loading...
          </Text>
</View>,
        "_debugID": 4,
        "_renderedOutput": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#007AFF",
                              "flex": 1,
                              "justifyContent": "center",
                            }
          }>
          <Text
                    accessible={true}
                    allowFontScaling={true}
                    ellipsizeMode="tail">
                    Loading...
          </Text>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignInScreen
    isLoading={true}
    signIn={[Function]} />,
}
`;

exports[`test renders correctly when not loading 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View
    style={
        Object {
            "alignItems": "center",
            "backgroundColor": "#007AFF",
            "flex": 1,
            "justifyContent": "center",
          }
    }>
    <View>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#FFFFFF",
                    "fontSize": 50,
                    "margin": 5,
                    "textAlign": "center",
                  }
            }>
            Scout
        </Text>
        <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
                Object {
                    "color": "#FFFFFF",
                    "fontSize": 20,
                    "marginBottom": 50,
                    "textAlign": "center",
                  }
            }>
            Your voice companion
        </Text>
        <IconButton
            backgroundColor="#FFFFFF"
            borderRadius={5}
            color="#000000"
            name="google"
            onPress={[Function]}
            size={20}>
            Login with Google
        </IconButton>
    </View>
</View>,
  "nodes": Array [
    <View
      style={
            Object {
                  "alignItems": "center",
                  "backgroundColor": "#007AFF",
                  "flex": 1,
                  "justifyContent": "center",
                }
      }>
      <View>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#FFFFFF",
                              "fontSize": 50,
                              "margin": 5,
                              "textAlign": "center",
                            }
                  }>
                  Scout
            </Text>
            <Text
                  accessible={true}
                  allowFontScaling={true}
                  ellipsizeMode="tail"
                  style={
                        Object {
                              "color": "#FFFFFF",
                              "fontSize": 20,
                              "marginBottom": 50,
                              "textAlign": "center",
                            }
                  }>
                  Your voice companion
            </Text>
            <IconButton
                  backgroundColor="#FFFFFF"
                  borderRadius={5}
                  color="#000000"
                  name="google"
                  onPress={[Function]}
                  size={20}>
                  Login with Google
            </IconButton>
      </View>
</View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <SignInScreen
        isLoading={false}
        signIn={[Function]} />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": SignInScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "isLoading": false,
          "signIn": [Function],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#007AFF",
                              "flex": 1,
                              "justifyContent": "center",
                            }
          }>
          <View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#FFFFFF",
                                                  "fontSize": 50,
                                                  "margin": 5,
                                                  "textAlign": "center",
                                                }
                              }>
                              Scout
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#FFFFFF",
                                                  "fontSize": 20,
                                                  "marginBottom": 50,
                                                  "textAlign": "center",
                                                }
                              }>
                              Your voice companion
                    </Text>
                    <IconButton
                              backgroundColor="#FFFFFF"
                              borderRadius={5}
                              color="#000000"
                              name="google"
                              onPress={[Function]}
                              size={20}>
                              Login with Google
                    </IconButton>
          </View>
</View>,
        "_debugID": 2,
        "_renderedOutput": <View
          style={
                    Object {
                              "alignItems": "center",
                              "backgroundColor": "#007AFF",
                              "flex": 1,
                              "justifyContent": "center",
                            }
          }>
          <View>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#FFFFFF",
                                                  "fontSize": 50,
                                                  "margin": 5,
                                                  "textAlign": "center",
                                                }
                              }>
                              Scout
                    </Text>
                    <Text
                              accessible={true}
                              allowFontScaling={true}
                              ellipsizeMode="tail"
                              style={
                                        Object {
                                                  "color": "#FFFFFF",
                                                  "fontSize": 20,
                                                  "marginBottom": 50,
                                                  "textAlign": "center",
                                                }
                              }>
                              Your voice companion
                    </Text>
                    <IconButton
                              backgroundColor="#FFFFFF"
                              borderRadius={5}
                              color="#000000"
                              name="google"
                              onPress={[Function]}
                              size={20}>
                              Login with Google
                    </IconButton>
          </View>
</View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <SignInScreen
    isLoading={false}
    signIn={[Function]} />,
}
`;
